src/widgets/scrollbar.rs

```diff
@@ -0,0 +1,208 @@
+use crate::{
+    buffer::Buffer,
+    layout::Rect,
+    style::Style,
+    symbols::{block::FULL, line::DOUBLE_VERTICAL},
+    widgets::Widget,
+};
+
+/// Scrollbar widget for tui-rs library.
+///
+/// This widget can be used to display a scrollbar in a terminal user interface.
+#[derive(Default, Debug, Clone)]
+pub struct Scrollbar<'a> {
+    scroll: f64,
```

Comment:
I've updated the code to make it show the size relative to the "number of lines" that the scroll bar can scroll over.
Previously, `scroll` was a ratio of `position / num_elements`. Now, the widget requires a user to pass in both `position` and `num_elements` explicitly.

Reply:
I think it could be useful to have multiple approaches to this (like the gauge does - it allows ratio, percent, etc.)

I'm not sure that I understand what num_elements represents just from the name:
1. The number of lines of the bar part of the scrollbar 
2. The total number of elements that are being scrolled (e.g. 1000 list items)
3. The total number of lines in the elements that are being scrolled (e.g. 1000 list items * 3 lines each)
4. The number of elements that are currently being shown (10 list items)
5. The number of lines that are currently being shown (30 lines)